orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTaskSpec.groovy

```diff
@@ -55,14 +58,16 @@ class DetermineHealthProvidersTaskSpec extends Specification {
     (taskResult.getStageOutputs() as Map) == expectedStageOutputs
 
     where:
-    stageContext                                  | application            || expectedStageOutputs
-    [interestingHealthProviderNames: []]          | null                   || [:]
-    [interestingHealthProviderNames: null]        | null                   || [:]
-    [application: "app"]                          | bA("app", true, false) || [interestingHealthProviderNames: ["Amazon"]]
-    [application: "app", cloudProvider: "gce"]    | bA("app", true, false) || [interestingHealthProviderNames: ["Google"]]
-    [application: "app"]                          | bA("app", true, true)  || [:]                                           // no health provider names when platformHealthOnlyShowOverride is true
-    [application: "app", cloudProvider: "random"] | null                   || [:]                                           // no health provider names when cloud provider is unsupported/unknown
-    [application: "app"]                          | null                   || [:]                                           // no health provider names when an exception is raised
+    stageContext                                                                                    | application               || expectedStageOutputs
+    [interestingHealthProviderNames: []]                                                            | null                      || [:]
+    [interestingHealthProviderNames: null]                                                          | null                      || [:]
+    [application: "app"]                                                                            | bA("app", true, false)    || [interestingHealthProviderNames: ["Amazon"]]
+    [application: "app", cloudProvider: "gce"]                                                      | bA("app", true, false)    || [interestingHealthProviderNames: ["Google"]]
+    [application: "app"]                                                                            | bA("app", true, true)     || [:]                                           // no health provider names when platformHealthOnlyShowOverride is true
+    [application: "app", cloudProvider: "titus"]                                                    | bA("app", true, true)     || [interestingHealthProviderNames: ["Titus"]]
+    [application: "app", cloudProvider: "titus", labels: [interestingHealthProviderNames: "Titus"]] | bA("app", false, true)    || [interestingHealthProviderNames: ["Titus"]]
```

Comment:
this ``` String.valueOf().equalsIgnoreCase()``` not legitimate  as in not necessary or made up? the label field is a [String:Object] dictionary. The string.valueOf() was meant to do the job of the toString() on the object to the the comparison against the string. I am happy to rewrite that, no big deal